baybe/insights/shap.py

```diff
@@ -334,6 +368,7 @@ def plot(
         data: pd.DataFrame | None = None,
         /,
         *,
+        target_index: int = 0,
```

Comment:
Hm, I think having 0 as default is a bit weird, since it somehow goes against the concept `default = most obvious / least surprising set`, i.e in this case 0 is just as (un-)reasonable as any other number. But I can see that you don't want to require users to pass anything in case of single-output models. So why not make it mandatory only for the multi-output setting, by using `None` as default and then asserting that it's provided by the user in multi-output contexts?